src/librustc_middle/middle/stability.rs

```diff
@@ -377,21 +383,39 @@ impl<'tcx> TyCtxt<'tcx> {
     /// Additionally, this function will also check if the item is deprecated. If so, and `id` is
     /// not `None`, a deprecated lint attached to `id` will be emitted.
     pub fn check_stability(self, def_id: DefId, id: Option<HirId>, span: Span) {
+        self.check_stability_internal(def_id, id, span, true, |span, def_id| {
```

Comment:
Why is this extra method necessary?

Reply:
TLDR: many unfixable errors are produced if we call `self.sess.delay_span_bug(span, &format!("encountered unmarked API: {:?}", def_id));` on generic parameters.

Generic parameters are not always marked stable.
This occurs when a generic from a dependency is used in a `staged_api` crate.
I believe it's caused by the parameter being instantiated in their users/callers crate rather than at the definition site.

Reply:
Hmm, so the issue is that generic parameters are the only language feature that has optional stability: everything else is either not a `staged_api` item, or is either explicitly marked stable or unstable. Maybe we could call this function `check_optional_stability` or something like that, and give the reason why we want it (i.e. specifically mention the case of generic parameters).